src/main/java/com/facebook/openwifirrm/optimizers/tpc/MeasurementBasedApApTPC.java

```diff
@@ -253,13 +288,18 @@ public Map<String, Map<String, Integer>> computeTxPowerMap() {
 				coverageThreshold,
 				nthSmallestRssi
 			);
+			logger.debug("  Old tx_power: {}", currentTxPower);
 			logger.debug("  New tx_power: {}", newTxPower);
-
-			Map<String, Integer> radioMap = new TreeMap<>();
-			radioMap.put(UCentralConstants.BAND_5G, newTxPower);
-			txPowerMap.put(serialNumber, radioMap);
+			txPowerMap.computeIfAbsent(serialNumber, k -> new TreeMap<>()).put(band, newTxPower);
 		}
+	}
 
+	@Override
+	public Map<String, Map<String, Integer>> computeTxPowerMap() {
```

Comment:
good to add a Javadoc since this is public

Reply:
either this method or the inner one should specify that this is the core of the TPC algorithm

Reply:
This is overriding a parent class method, which already has a javadoc. I added a mention that this is the core method in the javadoc in the parent class.